packages/tldraw/src/lib/shapes/arrow/elbow/routes/routeArrowWithAutoEdgePicking.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+	ElbowArrowRoute,
+	ElbowArrowSide,
+	ElbowArrowSideReason,
+	ElbowArrowSides,
+} from '../definitions'
+import { ElbowArrowWorkingInfo } from './ElbowArrowWorkingInfo'
+import { tryRouteArrow } from './elbowArrowRoutes'
+
+export function routeArrowWithAutoEdgePicking(info: ElbowArrowWorkingInfo): ElbowArrowRoute | null {
+	let idealRoute = null
+	if (Math.abs(info.gapX) > Math.abs(info.gapY) && info.midX !== null) {
+		if (info.gapX > 0) {
+			idealRoute = tryRouteArrow(info, 'right', 'left')
+		} else {
+			idealRoute = tryRouteArrow(info, 'left', 'right')
+		}
+	} else {
+		const aRight = info.A.edges.right
+		const aLeft = info.A.edges.left
+		const bTop = info.B.edges.top
+		const bBottom = info.B.edges.bottom
+
+		if (info.A.isPoint && info.B.isPoint) {
+			if (info.gapY > 0) {
+				idealRoute = tryRouteArrow(info, 'bottom', 'top')
+			} else {
+				idealRoute = tryRouteArrow(info, 'top', 'bottom')
+			}
+		} else if (
+			aRight &&
+			bTop &&
+			(aRight.expanded ?? aRight.value) <= bTop.crossTarget &&
+			aRight.crossTarget <= (bTop.expanded ?? bTop.value)
+		) {
+			idealRoute = tryRouteArrow(info, 'right', 'top')
+		} else if (
+			aRight &&
+			bBottom &&
+			(aRight.expanded ?? aRight.value) <= bBottom.crossTarget &&
+			aRight.crossTarget >= (bBottom.expanded ?? bBottom.value)
+		) {
+			idealRoute = tryRouteArrow(info, 'right', 'bottom')
+		} else if (
+			aLeft &&
+			bTop &&
+			(aLeft.expanded ?? aLeft.value) >= bTop.crossTarget &&
+			aLeft.crossTarget <= (bTop.expanded ?? bTop.value)
+		) {
+			idealRoute = tryRouteArrow(info, 'left', 'top')
+		} else if (
+			aLeft &&
+			bBottom &&
+			(aLeft.expanded ?? aLeft.value) >= bBottom.crossTarget &&
+			aLeft.crossTarget >= (bBottom.expanded ?? bBottom.value)
+		) {
+			idealRoute = tryRouteArrow(info, 'left', 'bottom')
+		} else if (info.gapY > 0) {
+			idealRoute = tryRouteArrow(info, 'bottom', 'top')
+		} else if (info.gapY < 0) {
+			idealRoute = tryRouteArrow(info, 'top', 'bottom')
+		}
+	}
+
+	if (idealRoute) {
+		idealRoute.aEdgePicking = 'auto'
+		idealRoute.bEdgePicking = 'auto'
+		return idealRoute
+	}
+
+	const aNonPartialSides = ElbowArrowSides.filter(
+		(side) => info.A.edges[side] && !info.A.edges[side]!.isPartial
+	)
+	const bNonPartialSides = ElbowArrowSides.filter(
+		(side) => info.B.edges[side] && !info.B.edges[side]!.isPartial
+	)
+
+	const nonPartialRouteCandidates = aNonPartialSides.flatMap((aSide) =>
+		bNonPartialSides.map((bSide) => [aSide, bSide, 'auto', 'auto'] as const)
+	)
+
+	return pickShortest(info, nonPartialRouteCandidates)
+}
+
+export function routeArrowWithPartialEdgePicking(
+	info: ElbowArrowWorkingInfo,
+	aSide: ElbowArrowSide
+) {
+	switch (aSide) {
+		case 'top':
+			return pickShortest(info, [
+				['top', 'bottom', 'manual', 'auto'],
```

Comment:
for these verbose things my mind always thinks if there's a shorthand way to express it:
```
const edgeMapper = (side) => ([aSide, side, 'manual', 'auto'])
...
case 'bottom':
  const edges = ['top', 'right', 'left', 'bottom'].map(edgeMapper)
	return pickShortest(info, edges)
```

Reply:
it's shorter but i don't think it's clearer, i guess? I kind of don't mind these being a little verbose, less moving parts for my brain to think about